pkg/extensions/sync/sync_test.go

```diff
@@ -0,0 +1,1367 @@
+package sync_test
```

Comment:
by changing package name to sync, you can add lot more tests that will cover all internal functions.

Reply:
In case you need to test internal functions, add a separate package 'sync_internal_test', that should work without moving the tests inside 'sync'.